master/docs/manual/credentialmanagement.rst

```diff
@@ -0,0 +1,100 @@
+Credential Management
+=====================
+
+Definition
+----------
+
+Buildbot needs to send and/or execute commands requiring credentials.
+
+The goal is to store credentials used by Buildbot commands.
+
+Credential could not be stored in ``buildbot.config`` or ``master.config`` because the configuration is available in a source control management.
+
+Credential are stored in the Buildbot database.
+
+Credentials once stored are obfuscated.
+
+Credential case: SSH Key
+------------------------
+
+A key used to send a ssh command is usually stored in a ``ssh text file`` (like id_rsa) in the :envvar:`HOME` directory.
+
+The ssh key registering is done with 3 entities: a name (ssh key name), a value (ssh_key value) and a path (where the file is stored).
+
+Credential case: Password
+-------------------------
+
+A password could be use in a command by Buildbot.
+The password is stored with 2 entities: a ``name``, and a ``value``. No path is inserted.
+
+How to add, delete or modify new credentials
+--------------------------------------------
+
+A data API is available to create, delete and modify new credentials.
+
+The data API modifies the 3 entities:
+``NAME`` and ``VALUE`` (mandatory), ``PATH`` (not mandatory, as it is not used in the password case).
+
+A data API function helps to register any new credential in the database.
+
+    .. code-block:: bash
+
+        buildbot add-creds http://localhost:8020 --file ~/.ssh/id_rsa --destination '$HOME/.ssh/id_rsa' --name ssh_user
+        buildbot add-creds http://localhost:8020  --name userpassword --value mypassword
+
+A data API allows users to modify credential values
+
+    .. code-block:: bash
+
+        buildbot modify-creds http://localhost:8020 --file ~/.ssh/id_rsa --destination '$HOME/.ssh/id_rsa' --name ssh_user
```

Comment:
add and modify creds are the same. We should rather have a set-creds command which does both use cases